### PR TITLE
DEPR: deprecate Timedelta.resolution

### LINKED ISSUE #21344: pd.Timedelta.resolution is different from datetime.timedelta.resolution
`timedelta.resolution` is documented as:
> The smallest possible difference between non-equal timedelta objects, timedelta(microseconds=1).

while `pandas.Timedelta.resolution` is doc-string'd as:
>return a string representing the lowest resolution that we have
```
In [9]: pd.Timedelta(minutes=30).resolution
Out[9]: 'T'

In [10]: timedelta(minutes=30).resolution
Out[10]: datetime.timedelta(0, 0, 1)
```

I think this difference should be addressed by either:
1) Documenting the difference between both but also supporting `datetime.timedelta`'s notion of resolution in `Timedelta` (i.e. something that returns `Timedelta(nanoseconds=1)`)
2) Making `Timedelta.resolution` consistent with `timedelta.resolution` and creating a new property for "the lowest resolution" of a `Timedelta` instance

Thoughts? 

xref #21336

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #21344
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

AFAICT we don't have any tests specifically for Timedelta.resolution, so added one for reso_str.

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -950,7 +950,7 @@ cdef class _Timedelta(timedelta):
         return np.int64(self.value).view('m8[ns]')
 
     @property
-    def resolution(self):
+    def reso_str(self):
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx]: where did this name come from? maybe min_resolution_string? or just resolution_string is more informative

**Review by jreback — CHANGES_REQUESTED** [review]
do we have tests for datetime.resolution?

**Comment by jreback** [comment]
can you add a this in the deprecation issue

**Comment by jbrockmendel** [comment]
> do we have tests for datetime.resolution?

Doubtful.  Timestamp.resolution and NaT.resolution are both wrong in subtly different ways, planning to do in separate PR.

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx]: More or less chosen out of thin air.  I think we might use it in some of the internal functions.  resolution_string sounds better, will update

**Comment by jbrockmendel** [comment]
> can you add a this in the deprecation issue

will do

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jorisvandenbossche** [comment]
We also have a Timestamp.resolution, DatetimeIndex.resolution, TimedeltaIndex.resolution. Should any of those be changed accordingly?

**Comment by jbrockmendel** [comment]
> We also have a Timestamp.resolution, DatetimeIndex.resolution, TimedeltaIndex.resolution. Should any of those be changed accordingly?

Yes.  Also NaT.resolution.  I'll open a new issue.

**Pull request merged**